### PR TITLE
Basic dead-letter recipe

### PR DESCRIPTION
**Pull request by lc512k**
`make dead-letter` to go through the dead letter queue ~~and log without deleting
@commuterjoy just to explore that queue for now~~

This puts the message through the pipeline, and if everything passes, ~~reenqueues it~~ resends it to the spoor-api.

### CODE REVIEW

File: server/dead-letter.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+const argv = require('minimist')(process.argv.slice(2));
+const region = argv.region || 'eu';
+const config = require('./aws.js');
+const EventModel = require('./models').EventModel;
+
+AWS.config.update(config[region].aws);
+
+const sqs = new AWS.SQS();
+const sqsUrlIngest = config[region].sqs['dead-letter'];
+
```
Comment (commuterjoy) [inline server/dead-letter.js]: probably throw an error if they don't have the ARN set in SQS_INGEST_US_DEAD_LETTER?

**Comment by commuterjoy** [comment]
+1

Would be interested to be able to have something like this, where we can pluck a few messages off the queue (Eg, up to 5) and then halt the process.

`make dead-letter --max 5`

File: server/dead-letter.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	const message = wrappedMessage.ingest._raw;
+
+	logger.info('\n\n\n Reenqueuing:', message)
+
+	sns.publish({
+		TopicArn: snsUrl,
+		Message: JSON.stringify(message)
+	}, (err) => {
+		if (err) {
+			logger.info('Message NOT reenqueued to SNS', message.messageId, err);
```
Comment (lc512k) [inline server/dead-letter.js]: This is currently failing for me here with `InvalidParameter: Invalid parameter: TopicArn`, but looking at the spoor-api and http://docs.aws.amazon.com/sns/latest/cli/sns_publish.html I can't see why. Will work on it some more

Comment (commuterjoy) [inline server/dead-letter.js]: SDK docs here BTW ...

http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/SNS.html#publish-property

I can't see anything wrong with your code.

**Pull request merged**